src/test/java/com/contact/application/controller/ContactControllerTest.java

```diff
@@ -0,0 +1,136 @@
+package com.contact.application.controller;
+
+import com.contact.application.model.Contact;
+import com.contact.application.service.TestUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.Assert.*;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class ContactControllerTest {
+    private TestUtils testUtils;
+    private GetController getControllerMock = mock(GetController.class);
+    private PostController postControllerMock = mock(PostController.class);
+    private UpdateController updateControllerMock = mock(UpdateController.class);
+    private DeleteController deleteControllerMock = mock(DeleteController.class);
+
+    private ContactController contactController;
+    private List<Contact> expectedContactList;
+    private Contact expectedContact;
+    private String expectedContactId;
+    private String invalidId;
+    @Before
+    public void setUp() {
+        testUtils = new TestUtils();
+        contactController = new ContactController(getControllerMock, postControllerMock, updateControllerMock, deleteControllerMock);
+        expectedContactId = testUtils.getUUID();
+        expectedContact = testUtils.getRandomContact(expectedContactId);
+        expectedContactList = Collections.singletonList(expectedContact);
+        invalidId = "invalid";
+
+        when(getControllerMock.getAllContacts()).thenReturn(expectedContactList);
+        when(getControllerMock.getContactById(eq(expectedContactId))).thenReturn(Optional.of(expectedContact));
+        when(getControllerMock.getContactById(eq(invalidId))).thenReturn(Optional.empty());
+
+        when(postControllerMock.createContact(eq(expectedContact))).thenReturn(Optional.of(expectedContactId));
+        when(postControllerMock.createContact(eq(null))).thenReturn(Optional.empty());
+
+        when(updateControllerMock.updateContact(eq(expectedContact), eq(expectedContactId))).thenReturn(Optional.of(expectedContact));
+        when(updateControllerMock.updateContact(eq(expectedContact), eq(invalidId))).thenReturn(Optional.empty());
+
+        when(deleteControllerMock.deleteContact(eq(expectedContact), eq(expectedContactId))).thenReturn(Optional.of(expectedContact));
+        when(deleteControllerMock.deleteContact(eq(expectedContact), eq(invalidId))).thenReturn(Optional.empty());
+    }
+
+    @Test
+    public void getAllContactsTest_ReturnValidList_AndReturnCorrectCode() {
```

Comment:
Your test naming is inconsistent

Reply:
Jesus Christ, pick something more important, lol. I will make that change.